packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
```

Comment:
I'm not sure if this should be listed here or not, but definitely a risk for integration within SLE is that we'll now have two totally different implementations for how to convert a number into a formatted file size -- one for when the value is displayed in the table and one for when the value is displayed anywhere else (i.e. just using the existing Angular pipe).

Reply:
I can't think of any practical way to avoid that. I don't think we want to let the client provide a formatting function, or to export our formatting function for a client to use elsewhere. Luckily, I don't think our formatting will be so different from the Angular pipe that it will be a glaring problem.

Reply:
As proposed now, I think it will be a problem. We're either going to have different units or a different value with the same units compared to the existing pipe.

Reply:
True, we will need to change either the units or the factor used by the current pipe, as you point out in your other comment.

Reply:
I've added this concern to the section, as suggested.